src/Units/Unit.java

```diff
@@ -218,9 +218,19 @@ public boolean isFullySupplied()
     return isFull;
   }
 
+  public int getCost()
+  {
+    return this.CO.getCost(model);
+  }
+
   public int getRepairCost()
   {
-    return model.getRepairCost(new UnitContext(this));
+    UnitContext uc = new UnitContext(this);
+    for( UnitModifier mod : getModifiers() )
+      mod.modifyCost(uc);
```

Comment:
Does this mean temporary stat/cost changes also affect repair costs? Is that intended?

Reply:
There are no non-dynamic stat/cost changes anymore, so yes and yes.

Reply:
Are repair costs discounted on the carts when units are on sale? Is this "intended" because it's how we prefer things to work, or because they naturally work that way with this new architecture approach? I don't think repairs were discounted before.

Reply:
I can't imagine any way they wouldn't be. It affects COP charge accumulation in AW2. In any case, the only situation where it'd possibly be relevant is BBoat repairs - you can't activate your power before repairs happen, and it doesn't stay active for the following repair phase.
It's intended because I can't imagine it works differently on cart, though I haven't tested this specific case.
As far as our implementation, I don't think there's been any time where repairs weren't discounted any time units were discounted (Bear would be really bad otherwise, until the repair cost change).